zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/scaling/MarkPartitionBootstrappedProcessor.java

```diff
@@ -77,12 +93,29 @@ public void processDistributedCommand(final TypedRecord<ScaleRecord> command) {
     final var scalingKey = command.getKey();
     final var wasAlreadyBootstrapped = areAllPartitionsBootstrapped();
     stateWriter.appendFollowUpEvent(scalingKey, ScaleIntent.PARTITION_BOOTSTRAPPED, scaleUp);
+    // if the partition that has completed bootstrapping is the current one, resubscribe to all
+    // message start events and signals
+    if (scaleUp.getRedistributedPartitions().getFirst() == command.getPartitionId()) {
+      subscribeToStartEventsAndSignals();
+    }
     if (!wasAlreadyBootstrapped && areAllPartitionsBootstrapped()) {
       stateWriter.appendFollowUpEvent(scalingKey, ScaleIntent.SCALED_UP, scaleUp);
     }
     distributionBehavior.acknowledgeCommand(command);
   }
 
+  private void subscribeToStartEventsAndSignals() {
+    processState.forEachProcess(
+        null,
+        persistedProcess -> {
+          final var deployed =
+              processState.getLatestProcessVersionByProcessId(
+                  persistedProcess.getBpmnProcessId(), persistedProcess.getTenantId());
+          startEventSubscriptions.resubscribeToStartEvents(deployed);
+          return true;
+        });
+  }
+
```

Comment:
:upside_down_face: Might make sense to introduce a new method to just iterate over the latest versions of all processes, that way we could avoid iterating over thousands of old versions.